crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -210,11 +214,18 @@ impl WorkerPool {
                 //
                 // Note: Rng::gen rolls between [0, 1.0) for f32, so this works as expected.
                 let failed = rand::thread_rng().gen::<f32>() < self.failure_rate;
+                let store = self.store.clone();
 
                 async move {
                     tracing::debug!("Begin proving batch.");
 
-                    let batch = Self::build_batch(transactions).map_err(|err| (id, err))?;
+                    let inputs = store
+                        .get_batch_inputs(
+                            transactions.iter().flat_map(|tx| tx.unauthenticated_notes()),
+                        )
+                        .await
+                        .map_err(|err| (id, err.into()))?;
+                    let batch = Self::build_batch(transactions, inputs).map_err(|err| (id, err))?;
```

Comment:
I forget now, but do we do something similar when constructing a block? Basically, curious what happens if we while we are building a batch one of the unauthenticated notes becomes authenticated (i.e., because the block in which it gets recorded gets committed right after this code is executed).